### PR TITLE
Feature/module find cmd full path

### PR DESCRIPTION
**Pull request by baberlevi**
Adding an option to get the filesystem path of the module file for a package.  Open to a different argument flag if you want it to follow some convention.  I considered also adding an argument to the `spack location` command, but thought it may be confusing since there is already a `--module-dir` argument there that means something totally different.

### CODE REVIEW

**Comment by baberlevi** [comment]
@adamjstewart @tgamblin  - Just checking to see if anyone has had a chance to look at this ?  It's my first PR to spack core, so let me know if you want something done differently, or a unit test added or something.

**Comment by adamjstewart** [comment]
LGTM. Pinging @alalazo who maintains a lot of the modules stuff.

**Comment by baberlevi** [comment]
Merged devel in to kick off travis again - passed. @alalazo  can you please take a look when you're able ?

File: lib/spack/spack/cmd/module.py [hunk 1]
```diff
@@ -208,8 +213,11 @@ def find(module_types, specs, args):
         msg += 'no {0} module has been generated for it.'
         tty.die(msg.format(module_type, spec))
 
-    # ... and if it is print its use name
-    print(writer.layout.use_name)
+    # ... and if it is print its use name or full-path if requested
+    if args.full_path:
+        print(writer.layout.filename)
```
Comment (alalazo) [inline lib/spack/spack/cmd/module.py]: To add a test case for this, you just need to change:
```python
def test_find(database, parser):
    """Tests the 'spack module find' under a few common scenarios."""

    # Try to find it for tcl module files
    args = parser.parse_args(['find', '--module-type', 'tcl', 'libelf'])
    module.module(parser, args)
```
in `lib/spack/spack/test/cmd/module.py` with:
```python
@pytest.mark.parametrize('cli_args', [
    ['--module-type', 'tcl', 'libelf'],
    ['--module-type', '--full-path', 'tcl', 'libelf']
])
def test_find(database, parser, cli_args):
    """Tests the 'spack module find' under a few common scenarios."""

    # Try to find it for tcl module files
    args = parser.parse_args(['find'] + cli_args)
    module.module(parser, args)
```

Comment (baberlevi) [reply lib/spack/spack/cmd/module.py]: Added the test.

Our use case: we are building packages centrally on a build server, and then syncing to a rsync/nfs round-robin pair for production use around our University. Having this option lets our sync script easily find the location for the lmod file to be synced (using spack location -i for the app)

**Comment by alalazo** [comment]
@baberlevi Waiting for Travis to merge, thanks!

**Pull request merged**